runtime-sdk/modules/evm/src/evm_backend.rs

```diff
@@ -166,82 +148,45 @@ impl<'c, C: oasis_runtime_sdk::Context> ApplyBackend for Backend<'c, C> {
                     reset_storage,
                 } => {
                     let addr: H160 = address.into();
-                    let is_empty = {
+                    // Derive SDK account address from the Ethereum address.
+                    let address = Cfg::map_address(address);
+
+                    // Update account balance and nonce.
+                    let mut state = self.ctx.get_mut().runtime_state();
+                    let amount =
+                        token::BaseUnits::new(basic.balance.as_u128(), Cfg::TOKEN_DENOMINATION);
```

Comment:
these panic if the u256 doesn't fit. maybe there are total-supply-related reasons why this wouldn't happen. although I guess the api of this backend doesn't let us return an error anyway

Reply:
This is correct, since everything is now backed by accounts that use `u128` this can never actually fail unless the EVM can create tokens out of thin air (and in this case the whole module is seriously broken).

Reply:
that the sdk total supply is u128, yeah